covenant-signer/signerservice/middlewares/hmac_auth.go

```diff
@@ -98,11 +102,17 @@ func ValidateHMAC(hmacKey string, body []byte, receivedHMAC string) (bool, error
 
 // RewindRequestBody reads a request body and then rewinds it, so it can be read again
 func RewindRequestBody(reader io.ReadCloser) ([]byte, io.ReadCloser, error) {
-	body, err := io.ReadAll(reader)
+	// The extra byte helps us determine if the body exceeds the limit
+	limitedReader := io.LimitReader(reader, MaxRequestBodySize+1)
```

Comment:
hmm we already have `r.Use(middlewares.ContentLengthMiddleware(int64(cfg.ServerConfig.MaxContentLength)))` middleware in our http middleware path before HMAC:
```
r.Use(middlewares.TracingMiddleware)
r.Use(middlewares.LoggingMiddleware)
r.Use(middlewares.ContentLengthMiddleware(int64(cfg.ServerConfig.MaxContentLength)))
// TODO: TLS configuration if server is to be exposed over the internet, if it supposed to
// be behind some reverse proxy like nginx or cloudflare, then it's not needed.
// Probably it needs to be configurable

// Add HMAC authentication middleware if configured
if cfg.ServerConfig.HMACKey != "" {
log.Info().Msg("HMAC authentication enabled")
r.Use(middlewares.HMACAuthMiddleware(cfg.ServerConfig.HMACKey))
```
do you think we need one here ?

Reply:
Good point I missed that, most probably we don't need it as it will just bloat implementation, will close this PR then.